controller_manager:
  ros__parameters:
    forward_position_controller:
      type: forward_command_controller/ForwardCommandController

    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    cartesian_motion_controller:
      type: cartesian_motion_controller/CartesianMotionController

    motion_control_handle:
      type: cartesian_controller_handles/MotionControlHandle

forward_position_controller:
  ros__parameters:
    joints:
      - shoulder_pan_joint
      - shoulder_lift_joint
      - elbow_joint
      - wrist_1_joint
      - wrist_2_joint
      - wrist_3_joint

joint_trajectory_controller:
  ros__parameters:
    joints:
      - shoulder_pan_joint
      - shoulder_lift_joint
      - elbow_joint
      - wrist_1_joint
      - wrist_2_joint
      - wrist_3_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
    state_publish_rate: 100.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    # constraints:
    #   stopped_velocity_tolerance: 0.2
    #   goal_time: 0.0
    #   shoulder_pan_joint: { trajectory: 0.2, goal: 0.1 }
    #   shoulder_lift_joint: { trajectory: 0.2, goal: 0.1 }
    #   elbow_joint: { trajectory: 0.2, goal: 0.1 }
    #   wrist_1_joint: { trajectory: 0.2, goal: 0.1 }
    #   wrist_2_joint: { trajectory: 0.2, goal: 0.1 }
    #   wrist_3_joint: { trajectory: 0.2, goal: 0.1 }

cartesian_motion_controller:
  ros__parameters:

    # See the cartesian_compliance_controller
    end_effector_link: "tcp_link"
    robot_base_link: "aubo_i5_base_link"
    joints:
      - shoulder_pan_joint
      - shoulder_lift_joint
      - elbow_joint
      - wrist_1_joint
      - wrist_2_joint
      - wrist_3_joint

    # See the cartesian_compliance_controller
    command_interfaces:
      - position
        #- velocity

    solver:
        error_scale: 1.0
        iterations: 10
        publish_state_feedback: True

    pd_gains:
        trans_x: {p: 10.0}
        trans_y: {p: 10.0}
        trans_z: {p: 10.0}
        rot_x: {p: 1.0}
        rot_y: {p: 1.0}
        rot_z: {p: 1.0}

motion_control_handle:
  ros__parameters:
    end_effector_link: "tcp_link"
    robot_base_link: "aubo_i5_base_link"
    # ft_sensor_ref_link: "tool0"
    joints:
      - shoulder_pan_joint
      - shoulder_lift_joint
      - elbow_joint
      - wrist_1_joint
      - wrist_2_joint
      - wrist_3_joint